DOCSP-39983: Add an optional bool flag to the write function to skip writing null fields (#21)
mongodb/docs-pymongo-arrow: source/quick-start.txt

   write(coll, arrow_table)
   write(coll, ndarrays)

.. note::
  
   NumPy arrays are specified as ``dict[str, ndarray]``.

Ignore Empty Values
~~~~~~~~~~~~~~~~~~~

The ``write()`` method optionally accepts a boolean ``exclude_none`` parameter.
If you set this parameter to ``True``, the driver doesn't write empty values to
the database. If you set this parameter to ``False`` or leave it blank, the 
driver writes ``None`` for each empty field.

> The code in the following example writes an Arrow ``Table`` to MongoDB twice. One
> of the values in the ``b`` field is set to ``None``. 
> 
> The first call to the ``write()`` method omits the ``exclude_none`` parameter, 
> so it defaults to ``False``. All values in the ``Table``, including ``None``, 
> are written to the database. The second call to the ``write()`` method sets 
> ``exclude_none`` to ``True``, so the empty value in the ``b`` field is ignored.

Reviewer: S: Because `b` is a field name and shown as a string in the output, do the same here

```suggestion
The code in the following example writes an Arrow ``Table`` to MongoDB twice. One
of the values in the ``'b'`` field is set to ``None``. 

The first call to the ``write()`` method omits the ``exclude_none`` parameter, 
so it defaults to ``False``. All values in the ``Table``, including ``None``, 
are written to the database. The second call to the ``write()`` method sets 
``exclude_none`` to ``True``, so the empty value in the ``'b'`` field is ignored.
```

   write(coll, arrow_table)
   write(coll, ndarrays)

> You can add an optional argument to ``write()`` called ``exclude_none``.

Reviewer: S: Maybe put this in a subheading?

```suggestion
Ignore Empty Values
~~~~~~~~~~~~~~~

You can add an optional argument to ``write()`` called ``exclude_none``.
```

> You can add an optional argument to ``write()`` called ``exclude_none``.

Reviewer: S: add a little more clarity

```suggestion
The ``write()`` method optionally accepts a boolean ``exclude_none`` parameter.
```

> This argument is ``False`` by default. When it is set to ``True``, empty
> values will not be written to the database.

Reviewer: S: active voice + second person

```suggestion
If you pass ``True`` for this parameter, the driver doesn't write empty
fields to the database. If you pass ``False`` or leave the parameter blank,
the driver writes the value ``None`` for each empty field.
```

> The code in the following example writes an Arrow ``Table`` to MongoDB twice: once
> with ``exclude_none`` defaulting to ``False``, and once with it set to ``True``.

Reviewer: Love this example code. I would be extra clear about what's happening where. Something like this:

```suggestion
The following code example writes an Arrow ``Table`` to MongoDB twice. The
first call to the ``write()`` method omits the ``exclude_none`` parameter,
so the empty field ``'b'`` is written to the database. The second call to the
``write()`` method sets ``exclude_none=True``, so the empty field ``'b'``
is ignored.
```

.. io-code-block::
   :copyable: true

>    .. input:: 
>       :language: python
>       :dedent:

Reviewer: S: Highlighting is great. (also don't think you need `dedent`)

```suggestion
   .. input:: 
      :language: python
      :emphasize-lines: 11, 15
```

      
      data_a = [1,2, 3, 4, 5, 6]

>       data_b = [1, 2, 3, None, 5, 6]

Reviewer: Q: Does `exclude_none` ignore the entire `data_b` array because it contains a single `None` value? If so, might want to reword some things earlier

Author: No, it just ignores the one ``None`` value. I will make the example more clear that this is the case
